### PR TITLE
sprint2 task:-superadmin login and superadmin dashboard

### PR DESCRIPTION
**Pull request by Pkjavaqwe**
super admin login and dashboard

### CODE REVIEW

File: src/app/components/header/header.component.ts [hunk 1]
```diff
@@ -1,9 +1,10 @@
 import { Component } from '@angular/core';
+import { NavbarComponent } from "../navbar/navbar.component";
 
 @Component({
   selector: 'app-header',
   standalone: true,
-  imports: [],
+  imports: [NavbarComponent],
```
Comment (neosoftshankarlad) [inline src/app/components/header/header.component.ts]: Navbar Component is separately created so, why need this header component ,
you can directly render the navbar component ?

File: src/app/components/index.ts
Comment (neosoftshankarlad) [inline src/app/components/index.ts:1]: Remove the file if it empty or not in use.

File: src/app/components/organizationcard/organizationcard.component.css [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+      max-width: 45%;
+    }
+  }
+  
+  @media (min-width: 1024px) {
+    .customer-card {
+      max-width: 30%;
+    }
+  }
+  */
```
Comment (neosoftshankarlad) [inline src/app/components/organizationcard/organizationcard.component.css]: Remove the commented code from the files.

File: src/app/components/super-admin-login/super-admin-login.component.css [hunk 3]
```diff
@@ -0,0 +1,39 @@
+html,
+body {
+  height: 100%;
+  display: flex;
+  /* flex-direction: column; */
```
Comment (neosoftshankarlad) [inline src/app/components/super-admin-login/super-admin-login.component.css]: remove the commented lines. every where from the files.

File: src/app/models/user.ts
Comment (neosoftshankarlad) [inline src/app/models/user.ts:1]: remove the files if it is not used.,

File: src/app/services/organization.service.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { Organization } from '../models/organizations';
+import { Observable } from 'rxjs';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class OrganizationService {
+  baseUrlOrg="http://localhost:5000/api/organizations"
```
Comment (neosoftshankarlad) [inline src/app/services/organization.service.ts]: Use environment.ts file to declare the URL and global y required parameter throughout the app.

File: src/app/app.routes.ts [hunk 5]
```diff
@@ -1,3 +1,19 @@
...
+import { SuperAdminLoginComponent } from './components/super-admin-login/super-admin-login.component';
+import { SuperadminComponent } from './components/superadmin/superadmin.component';
+import { authGuard } from './guards/authguard.guard';
 
-export const routes: Routes = [];
+export const routes: Routes = [
+  {
+    path: '',
+    component: SuperAdminLoginComponent,
+    // loadComponent:()=>import("./components/super-admin-login/super-admin-login.component").then(m=>m.SuperAdminLoginComponent)
```
Comment (anit-neosoft) [inline src/app/app.routes.ts]: remove comments

File: src/app/app.routes.ts [hunk 6]
```diff
@@ -1,3 +1,19 @@
...
+   
+  },
+  {
+    path: 'superAdminDashboard/:id',
+    // component: SuperadminComponent,
```
Comment (anit-neosoft) [inline src/app/app.routes.ts]: remove comment

File: src/app/authguard.guard.spec.ts
Comment (anit-neosoft) [inline src/app/authguard.guard.spec.ts:1]: delete the file

File: src/app/authguard.guard.ts
Comment (anit-neosoft) [inline src/app/authguard.guard.ts:1]: What is this file for? @Pkjavaqwe @kailaskg

File: src/app/components/header/header.component.css
Comment (anit-neosoft) [inline src/app/components/header/header.component.css:1]: Remove

File: src/app/components/index.ts
Comment (anit-neosoft) [inline src/app/components/index.ts:1]: Delete

File: src/app/components/organizationcard/organizationcard.component.spec.ts
Comment (anit-neosoft) [inline src/app/components/organizationcard/organizationcard.component.spec.ts:1]: delete

File: src/app/components/organizationcard/organizationcard.component.css
Comment (anit-neosoft) [inline src/app/components/organizationcard/organizationcard.component.css:1]: Make the styles global to keep the theme consistent

File: src/app/components/organizationcard/organizationcard.component.html
Comment (anit-neosoft) [inline src/app/components/organizationcard/organizationcard.component.html:1]: Change the card layout to show less data on the card and show the card details elsewhere

File: src/app/components/organizationcard/organizationcard.component.ts
Comment (anit-neosoft) [inline src/app/components/organizationcard/organizationcard.component.ts:1]: change the folder and component name

File: src/app/components/super-admin-login/super-admin-login.component.css [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+}
+.login {
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  justify-content: center;
+  color: rgb(19, 20, 20);
+  width: 300px;
+}
+.butn {
```
Comment (anit-neosoft) [inline src/app/components/super-admin-login/super-admin-login.component.css]: change to .button name the classes with complete names

File: src/app/components/super-admin-login/super-admin-login.component.css [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+  margin-top: 8px;
+  justify-content: center;
+}
+mat-card {
```
Comment (anit-neosoft) [inline src/app/components/super-admin-login/super-admin-login.component.css:17]: make the styles global refer to the DRY principle

File: src/app/components/super-admin-login/super-admin-login.component.html [hunk 9]
```diff
@@ -0,0 +1,68 @@
+<!DOCTYPE html>
```
Comment (anit-neosoft) [inline src/app/components/super-admin-login/super-admin-login.component.html]: remove the html boilerplate

File: src/app/components/super-admin-login/super-admin-login.component.html [hunk 10]
```diff
@@ -0,0 +1,68 @@
...
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <title>Document</title>
+  </head>
+  <body>
+    <mat-card>
+      <form [formGroup]="loginForm" (submit)="loginAdmin()">
+        <div class="login">
+          <p>SuperAdmin Login</p>
```
Comment (anit-neosoft) [inline src/app/components/super-admin-login/super-admin-login.component.html]: use `<mat-card-header>`

File: src/app/components/super-admin-login/super-admin-login.component.html [hunk 11]
```diff
@@ -0,0 +1,68 @@
...
+              </mat-error>
+              <mat-error *ngIf="noSpecialChars">
+                at least one special character <strong>required</strong>
+              </mat-error>
+              <mat-error *ngIf="noLowerCase">
+                at least one lowercase character <strong>required</strong>
+              </mat-error>
+              <mat-error *ngIf="noUpperCase">
+                at least one upperCase character <strong>required</strong>
+              </mat-error>
```
Comment (anit-neosoft) [inline src/app/components/super-admin-login/super-admin-login.component.html]: refer to https://stackoverflow.com/questions/59764694/handling-with-multiple-mat-errors

File: src/app/components/super-admin-login/super-admin-login.component.ts [hunk 12]
```diff
@@ -0,0 +1,110 @@
...
+  errorMessage: string = '';
+  accessToken: Token = {
+    token: '',
+    message: '',
+    statuscode: 0,
+    success: false,
+    _id: '',
+  };
+  passwordValidity: string = '';
+  StrongPasswordRegx: RegExp = /^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)[a-zA-Z\d]{8,}$/;
```
Comment (anit-neosoft) [inline src/app/components/super-admin-login/super-admin-login.component.ts]: maintain camel case when naming variables

File: src/app/components/super-admin-login/super-admin-login.component.ts [hunk 13]
```diff
@@ -0,0 +1,110 @@
...
+    };
+    console.log('outside if loginAdmin');
+    console.log('loginCred-', login);
+    if (login.password && login.email) {
+      const tokenObservable = this.authService.authenticateLogin(login);
+      console.log('tokenObservable', tokenObservable);
+      tokenObservable.subscribe({
+        next: (data) => {
+          console.log('token', data);
+          this.accessToken = data;
```
Comment (anit-neosoft) [inline src/app/components/super-admin-login/super-admin-login.component.ts]: there is no need to create extra variable accessToken rather, just give type to data and access the data object directly.

File: src/app/components/superadmin/superadmin.component.spec.ts
Comment (anit-neosoft) [inline src/app/components/superadmin/superadmin.component.spec.ts:1]: delete file

File: src/app/customValidators/custom-password-validators.spec.ts
Comment (anit-neosoft) [inline src/app/customValidators/custom-password-validators.spec.ts:1]: delete this

File: src/app/guards/authguard.guard.spec.ts
Comment (anit-neosoft) [inline src/app/guards/authguard.guard.spec.ts:1]: delete this

File: src/app/modules/organization/components/index.ts
Comment (anit-neosoft) [inline src/app/modules/organization/components/index.ts:1]: delete this

File: src/app/modules/organization/services/index.ts
Comment (anit-neosoft) [inline src/app/modules/organization/services/index.ts:1]: delete this

File: src/app/modules/retailer/components/index.ts
Comment (anit-neosoft) [inline src/app/modules/retailer/components/index.ts:1]: delete this

File: src/app/modules/retailer/services/index.ts
Comment (anit-neosoft) [inline src/app/modules/retailer/services/index.ts:1]: delete this

File: src/app/modules/user/components/index.ts
Comment (anit-neosoft) [inline src/app/modules/user/components/index.ts:1]: delete this

File: src/app/modules/user/services/index.ts
Comment (anit-neosoft) [inline src/app/modules/user/services/index.ts:1]: delete this

File: src/app/services/auth.service.ts [hunk 14]
```diff
@@ -1,9 +1,31 @@
...
+      environment.apiEndpointauth + '/login',
+      loginCredentials
+    );
+    return data;
+  }
+  isAuthenticated(): boolean {
+    const setToken = sessionStorage.getItem('token');
+    if (setToken) {
+      return true;
+      /*this.route.navigate([
```
Comment (anit-neosoft) [inline src/app/services/auth.service.ts]: remove comments

File: src/app/services/organization.service.spec.ts
Comment (anit-neosoft) [inline src/app/services/organization.service.spec.ts:1]: delete

File: src/app/services/organization.service.ts [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+
+@Injectable({
+  providedIn: 'root'
+})
+export class OrganizationService {
+  baseUrlOrg=environment.apiEndpointOrganization
+  constructor(private http:HttpClient) { }
+  getAllOrganizationsApi():Observable<allOrganization>{
+    // here extract data array and return 
+    const observableData =this.http.get<allOrganization>(this.baseUrlOrg+'/getall')
```
Comment (anit-neosoft) [inline src/app/services/organization.service.ts]: build the string in a const variable and pass it to the api call

File: src/environments/environment.development.ts
Comment (anit-neosoft) [inline src/environments/environment.development.ts:1]: Dont push environment files

File: src/environments/environment.ts
Comment (anit-neosoft) [inline src/environments/environment.ts:1]: Dont't push

File: src/app/components/super-admin-login/super-admin-login.component.spec.ts
Comment (anit-neosoft) [inline src/app/components/super-admin-login/super-admin-login.component.spec.ts:1]: Delete

**Review by anit-neosoft — CHANGES_REQUESTED** [review]
Make the changes as per the comments
- Remove console.logs()
- Remove comments
- Follow the naming conventions
- Create extra variables only if necessary.

File: src/app/authguard.guard.ts
Comment (Pkjavaqwe) [reply src/app/authguard.guard.ts:1]: not used was created without folder by mistake, now deleted

File: src/app/services/organization.service.ts (hunk 15 above)
Comment (Pkjavaqwe) [reply src/app/services/organization.service.ts]: built the string out of method getAllOrganizationsApi and then passed it to api call

**Pull request merged**